testnet/stacks-node/src/event_dispatcher.rs

```diff
@@ -1065,19 +1159,30 @@ impl EventDispatcher {
     pub fn process_new_stackerdb_chunks(
         &self,
         contract_id: QualifiedContractIdentifier,
-        new_chunks: Vec<StackerDBChunkData>,
+        modified_slots: Vec<StackerDBChunkData>,
     ) {
         let interested_observers = self.filter_observers(&self.stackerdb_observers_lookup, false);
 
-        if interested_observers.len() < 1 {
+        let interested_receiver = STACKER_DB_CHANNEL.is_active();
+        if interested_observers.is_empty() && !interested_receiver {
             return;
         }
 
-        let payload = serde_json::to_value(StackerDBChunksEvent {
+        let event = StackerDBChunksEvent {
             contract_id,
-            modified_slots: new_chunks,
-        })
-        .expect("FATAL: failed to serialize StackerDBChunksEvent to JSON");
+            modified_slots,
+        };
+        let payload = serde_json::to_value(&event)
+            .expect("FATAL: failed to serialize StackerDBChunksEvent to JSON");
+
+        if interested_receiver {
+            if let Err(send_err) = STACKER_DB_CHANNEL.sender.send(event) {
```

Comment:
Suggestion: implement `StackerDBChannel::send(..)` directly, instead of exposing the inner `sender`.